src/asm/parser.y

```diff
@@ -1096,7 +1096,15 @@ print_expr:
 		printf("$%" PRIX32, $1);
 	}
 	| string {
-		fputs($1.c_str(), stdout);
+		// Allow printing NUL characters
+		size_t n = $1.length();
+		for (size_t i = 0; i < n;) {
+			i += printf("%s", $1.c_str() + i);
```

Comment:
If this `printf` fails, we're in for some nice fireworks.

Thanks, C.

Reply:
(Note also that it may return *less* than the string length, if it's unable to write more. If this returned 0 forever, RGBASM would just hang. Not sure that's great? Honestly I'd just `fwrite($1.data(), $1.length(), 1, stdout)` that.)